### PR TITLE
Metadata

### PR DESCRIPTION
**Pull request by mgckind**

### CODE REVIEW

**Comment by mgckind** [comment]
@kadrlica  Can you take a look and do a quick review? Thanks!

**Comment by kadrlica** [comment]
I can try to take a look tonight.

**Comment by kadrlica** [comment]
Table tab completion is only working for my personal tables when I specify the owner prefix. For example `describe_table KADRLICA.MY_BL` tab completes but `describe_table MY_BL` does not.

**Comment by mgckind** [comment]
Thanks, it should be quick, just making sure it makes sense.
I basically create a CACHE_TABLE and CACHE_COLUMNS to keep the metadata in DES_ADMIN (on every DB) with Chri's help. Now the loading is MUCH faster. I have a script which can be run nightly to keep those tables up to date.

**Comment by mgckind** [comment]
The thing you mentioned about prefix, the query is actually this:
`select table_name from DES_ADMIN.CACHE_TABLES  
            union select table_name from user_tables`
That union should give you tab autocompletion to your own tables as well.

**Comment by kadrlica** [comment]
Thanks. Sorry, I removed that comment because I realized I was testing on an older version. I'll update and verify.

**Comment by kadrlica** [comment]
Ok, I've updated to `1.3.2-dev-fd41ecc` and it looks like things are behaving pretty well. I've submitted issue #94 to address some strange behavior with tab completing, which is probably outside of the scope of the PR.

I'm not seeing large improvements in load time. If the `FGOTTENMETADATA` table doesn't exist, it takes ~15s to load the cache on DESSCI for both `1.3.0` and `1.3.2-dev-fd41ecc`. When `FGOTTENMETADATA` does exist it takes ~9s on each.

File: easyaccess/easyaccess.py [hunk 1]
```diff
@@ -1370,49 +1362,25 @@ def do_set_password(self, arg):
...
         try:
-            qt = self.cur.execute(query_time)
-            last = qt.fetchall()
-            now = datetime.datetime.now()
-            diff = abs(now - last[0][0]).seconds / (3600.)
-            if verb: print('Updated %.2f hours ago' % diff)
-        except:
-            pass
-        try:
-            query = "DROP TABLE FGOTTENMETADATA"
```
Comment (kadrlica) [inline easyaccess/easyaccess.py:1400]: This should have been `DROP TABLE FGOTTENMETADATA PURGE`. It looks like it has been removed completely, but that makes me wonder whether `FGOTTENMETADATA` is ever updated automatically.

**Comment by mgckind** [comment]
Sorry,  my fault, I've should have update the bin folder (or you can run from the folder itself). You should definitely see a much faster loading, ~1 sec or so in total. The problem is (and I'll fix it soon) you probably were still using the easyaccess from the bin folder which wasn't updated. A way to check if to see whether you see the change_db command (which is recent as well).

I'll merge this PR and have the bin folder in sync with the latest changes, I'll create another branch to fix that  once for all

**Comment by mgckind** [comment]
I'd like to keep the FGOTTENMETADATA table (even an empty one), Its the only way we have right now to keep track of easyaccess users vs other users.

**Comment by kadrlica** [comment]
Thanks that change to the `bin/easyaccess` worked. 

I wasn't suggesting removing `FGOTTENMETADATA` just noting that we were removing it before and are no longer doing so. Because we were removing the table before, it would be recreated periodically. Now, I don't think it will ever be updated (but I'm probably missing something).

**Comment by mgckind** [comment]
yeah, its was a conflict type between different computers, I'll do it the proper way soon(ish).

Yes, we are not dropping it anymore, and now I have a try/except to create it every time (as empty table) to make sure every ea user has it there. 

Thanks for looking into this!

**Pull request merged**